lib/bap/bap.mli

```diff
@@ -6593,6 +6593,11 @@ module Std : sig
       target.  *)
   val register_target : arch -> (module Target) -> unit
 
+  (** [provide_bil_transformation name f] register a new function [f] with a [name]
+      that will be applied to [bil] once instruction lifted and (optionaly)
+      after other trnasformations. *)
+  val provide_bil_transformation : string -> (bil -> bil) -> unit
```

Comment:
We need to align this interface with the upcoming bap services and, at the same time, consistent with existing Project passes. 
1) move the function into the Bil module, and rename to `register_pass`
2) add a function `passes : unit -> pass list` 
3) add a function `select_passes : pass list -> unit`
4) add module `Bil.Pass` which implements the `Printable` interface.  

Documentation for the new functions (and the specifications)

```ocaml
(** [register_pass name pass] provides a pass to the BIL transformation pipeline. 
     
     The BIL transformation pipeline is applied after the lifting procedure, i.e., it is embedded into 
     each [lift] function of all Target modules. (You can selectively register_passes based on architecture
     by subscribing to the [Project.Info.arch] variable). All passes that  were in the selection provided to 
     the [select_passes] are applied in the order of the selection until the fixed point is reached or a loop 
     is detected. 
     By default, no passes are selected. The [bil] plugin provides a user interface for passes selection, 
     as well as some useful passes. 
*)
val register_pass : string -> (bil -> bil) -> unit

(** [select_passes passes] select the [passes] for the BIL transformation pipeline. 
     See {!register_pass} for more information about the BIL transformation pipeline *)
val select_passes : pass list -> unit